meta.go

```diff
@@ -241,16 +241,53 @@ func parseStringList(list interface{}) []string {
 	return result
 }
 
-var termNameRE = regexp.MustCompile("^[a-z]+([a-z0-9-]+)(/[0-9]+)?$")
+var (
+	revSnippet  = `(?:\/(\d+))`
+	termSnippet = `[a-zA-Z][\w-]`
+	//termNameRE  = regexp.MustCompile(fmt.Sprintf(`^(?:(%s+)\/)?((?:%s)+)%s?$`, termSnippet, termSnippet, revSnippet))
+	termNameRE = regexp.MustCompile(`^(?:([^\/\\][\w-]+)\/)?([a-zA-Z][\w-]+)(?:\/(\d+))?$`)
```

Comment:
Charm URLs validate the user component with names.IsValidUser. I think term ID validation should do the same for consistency.

A regex is fine to validate the general syntactical structure (slash-delimited fields with some basic constraints, etc), but it might be too hard to read and understand to perform _all_ the validation in the regex.